README.md

```diff
@@ -135,22 +137,14 @@ echo 8 > /sys/class/net/enp2s0f0/device/sriov_numvfs
 ### Main parameters
 * `name` (string, required): the name of the network
 * `type` (string, required): "sriov"
-* `master` (string, required): name of the PF
-* `l2enable` (boolean, optional): if `true` then add VF as L2 mode only, IPAM will not be executed
+* `deviceID` (string, required): A valid pci address of an SRIOV NIC's VF. e.g. "0000:03:02.3"
```

Comment:
why this is required , don't we get the DeviceID from multus?

Reply:
Yes, we get this from Multus. What happens if some reason this information is not given(by user or other plugin)? The plugin won't be able to work and is should fail. Remember, we no longer try to allocate a free VF in sriov-cni anymore.

Reply:
So, deviceID indicates the VF that will be passed to the container.
can you elaborate on why this is required in the CNI's configuration file ? does it mean that every container get the same deviceID?

The way the SRIOV-CNI plugin is designed today, to my understanding (please correct me if i'm wrong),
it is coupled with both SR-IOV device plugin and Multus.
at a high level : SR-IOV DP does the VF accounting  and Multus passes the DeviceID to SRIOV-CNI.

Is there a constellation where SRIOV-CNI can operate without Multus ?
Is there a constellation where SRIOV-CNI can operate without SRIOV DP ?

if not, then i feel we should document this limitation. (maybe add some docs on how to do a deployment?)

in the old implementation, we had a master PF for a given network, SR-IOV CNI did the discovery, allocation, configuration and move VF to container.

Reply:
``DeviceID`` is a required parameter for SR-IOV CNI to work, but it's not intended to be specified by user who creates the network CRD, rather rely on meta plugin (Multus) to pass along it. I think the reason why it's listed here (please correct me if I'm wrong) is to indicate that this is a required parameter, no matter who is responsible for passing or specifying it. Perhaps we can add a note after ``DeviceID`` saying that it's nowadays configured by Multus, no need for manual intervention.

Also for the old implementation that allocating VF randomly from a given PF, we have an issue tracking the deprecation of that mode #69 .

Re the documentation of how to do a deployment, maybe we can link to the SR-IOV DP doc which has covered instructions and steps or create a new section in the SR-IOV CNI README for easy consumption.

Reply:
@adrianchiris 
There are known limitation of managing and allocating SR-IOV VFs with sriov-cni on it's own. This has been discussed, reviewed and verified by community already. Please have a look at the this [doc](https://docs.google.com/document/d/1Ewe9Of84GkP0b2Q2PC0y9RVZNkN2WeVEagX9m99Nrzc/) (You will need to send access request) to find some answers to your question.

Reply:
@zshi-redhat  +1 On adding a note after DeviceID.
@ahalim-intel  Thanks for sharing the doc, I am aware there are limitations, i was just pointing out there is need for some documentation in the README on how to do a deployment that will work.
like zshi suggested maybe a link to : https://github.com/intel/sriov-network-device-plugin#quick-start